cmd/serve.go

```diff
@@ -1,23 +1,62 @@
 package cmd
 
 import (
+	"os"
+
+	"github.com/hashicorp/go-hclog"
+	hPlugin "github.com/hashicorp/go-plugin"
 	"github.com/odpf/optimus/cmd/server"
 	"github.com/odpf/optimus/config"
-	"github.com/odpf/salt/log"
+	"github.com/odpf/optimus/plugin"
 	cli "github.com/spf13/cobra"
 )
 
-func serveCommand(l log.Logger, conf config.Optimus) *cli.Command {
+func serveCommand() *cli.Command {
 	c := &cli.Command{
 		Use:     "serve",
 		Short:   "Starts optimus service",
 		Example: "optimus serve",
 		Annotations: map[string]string{
 			"group:other": "dev",
 		},
-		RunE: func(c *cli.Command, args []string) error {
-			return server.Initialize(l, conf)
-		},
 	}
+
+	c.RunE = func(c *cli.Command, args []string) error {
+		// TODO: find a way to load the config in one place
+		conf, err := config.LoadOptimusConfig()
+		if err != nil {
+			panic(err.Error())
+		}
+
+		// initiate jsonLogger
+		jsonLogger := initLogger(jsonLoggerType, conf.Log)
+
+		// initiate plugin log level
+		pluginLogLevel := hclog.Info
+		if conf.Log.Level == config.LogLevelDebug {
+			pluginLogLevel = hclog.Debug
+		}
+
+		// discover and load plugins. TODO: refactor this
```

Comment:
This can be integrated to server initialisation after the pr for server init is merged.